website/static/js/waterbutler.js

```diff
@@ -22,67 +21,74 @@ function getDefaultOptions(path, provider) {
     return options;
 }
 
-function buildUrl(suffix, path, provider, nid, options) {
-    path = path || '/';
-    var baseUrl = window.contextVars.waterbutlerURL + suffix;
-    return baseUrl + $.param($.extend(getDefaultOptions(path, provider), {nid: nid}, options));
-}
-
-var buildCrudUrl = buildUrl.bind(this, 'file?');
-var buildCopyUrl = buildUrl.bind(this, 'copy?');
-var buildMoveUrl = buildUrl.bind(this, 'move?');
-var buildMetadataUrl = buildUrl.bind(this, 'data?');
-var buildRevisionsUrl = buildUrl.bind(this, 'revisions?');
-var buildCreateFolderUrl = buildUrl.bind(this, 'folders?');
-
-
-function buildUploadUrl(path, provider, nid, file, options) {
-    path = (path || '/') + file.name;
-    return buildUrl('file?', path, provider, nid, options);
+function getViewOnly() {
+    return $osf.urlParams().view_only;
 }
 
-function buildFromTreebeard(suffix, item, options) {
-    return buildUrl(suffix, item.data.path, item.data.provider, item.data.nodeId, options);
+function buildFromTreebeard(item, options) {
+    return buildUrl(item.data.path, item.data.provider, item.data.nodeId, options);
 }
 
-function buildFromTreebeardFile(item, file, options) {
-    return buildUploadUrl(item.data.path, item.data.provider, item.data.nodeId, file, options);
+function addToken(options) {
+    var wb_options = $.extend({}, options || {}, {token: window.contextVars.accessToken});
+    return wb_options;
 }
 
-function toJsonBlob(item, options) {
-    return $.extend(getDefaultOptions(item.data.path || '/', item.data.provider), {nid: item.data.nodeId}, options);
+// Move obj.data.attributes contents to obj.data , for WB v1 API
+function wbLazyLoadPreprocess(obj) {
+    // WB v1 API now stores file info in data.attributes.
+    // Treebeard expects to find it in data
+    function _promoteAttrs(obj_data) {
```

Comment:
Sorry, I told you to declare this inside `wbLazyLoadPreprocess` and now I can't remember why.  The name is good, but can you move it outside `wbLazyLoadPreprocess`?  Also, can you expand on the comment?  Show what WB returns, and what format treebeard wants, so that devs who aren't familiar with tb/wb don't need to RTM to understand what this does.  Something like:

    // This function turns a WB structure into a TB structure
    // WB returns a structure like: data: { attributes: { name: 'foo', kind: 'file', ... } }
    // TB wants a structure like: data: { name: 'foo', kind: 'file', ... }